### PR TITLE
Add timeout and policy cache options to SDK methods

### PR DESCRIPTION
**Pull request by giarc3**
TODO: 
- [x] Depend on release ironoxide-java
- [x] Changelog

### CODE REVIEW

File: src/test/scala/ironoxide/v1/FullIntegrationTest.scala [hunk 1]
```diff
@@ -58,14 +59,26 @@ class FullIntegrationTest extends AsyncWordSpec with Matchers with EitherValues
       primaryTestUserSigningPrivateKeyBytes
     )
 
-  val sdk = IronOxide.initialize[IO](deviceContext).unsafeRunSync
+  val defaultConfig = IronOxideConfig()
+  val shortConfig = IronOxideConfig(PolicyCachingConfig(), Some(Duration(5, "millis")))
```
Comment (clintfred) [inline src/test/scala/ironoxide/v1/FullIntegrationTest.scala]: Is this the recommended Scala way of specifying the units? A quick google search indicated that `java.util.concurrent.TimeUnit.MILLISECONDS` might be an alternative.

File: src/main/scala/ironoxide/v1/common/IronOxideConfig.scala [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+      javaPolicy <- policyCaching.toJava
+      javaTimeout = sdkOperationTimeout.map(_.toJsdkDuration).orNull
+    } yield new jsdk.IronOxideConfig(javaPolicy, javaTimeout)
+}
+
+object IronOxideConfig {
+  def apply(): IronOxideConfig = {
+    val javaConfig = new jsdk.IronOxideConfig
+    val javaPolicy = javaConfig.getPolicyCachingConfig
+    val javaTimeout = javaConfig.getSdkOperationTimeout.toScala
```
Comment (clintfred) [inline src/main/scala/ironoxide/v1/common/IronOxideConfig.scala]: Curious: Where does this `toScala` come from?

Comment (giarc3) [reply src/main/scala/ironoxide/v1/common/IronOxideConfig.scala]: In "package.scala" we have `implicit class OptionalOps` for java's `Optional` that has a `toScala()` function to turn it into scala's `Option`.

File: src/test/scala/ironoxide/v1/FullIntegrationTest.scala (hunk 1 above)
Comment (giarc3) [reply src/test/scala/ironoxide/v1/FullIntegrationTest.scala]: I didn't see a recommended way, and both ways are referenced equally in the documentation. I originally used the string representation as it was more succinct, but I think I do prefer the `TimeUnit` way for the safety of using set values.
I changed it to `scala.concurrent.duration.MILLISECONDS`

**Pull request merged**